sdk/go/pulumi-language-go/main.go

```diff
@@ -184,9 +188,12 @@ func (cmd *mainCmd) Run(p *runParams) error {
 		cancel() // deregister handler so we don't catch another interrupt
 		close(cancelChannel)
 	}()
-	err = rpcutil.Healthcheck(ctx, p.engineAddress, 5*time.Minute, cancel)
-	if err != nil {
-		return fmt.Errorf("could not start health check host RPC server: %w", err)
+	if p.engineAddress == "" {
```

Comment:
Shouldn't this be != ""?

Reply:
I don't think so.  When we launch from pulumi cli we set this from an arg.  If it is empty we launched manually (presumably for debugging)

Reply:
Yes and if it's not empty we can hook healthchecks here, else we have to wait till handshake time. This check is "!=" in both the node and python hosts, its just here in Go you've got it flipped.

Reply:
This still looks wrong. Your checking for engineAddress=="" but then passing it as the connection string to Healthcheck, flip the if condition.